external/cdc/cds.go

```diff
@@ -0,0 +1,169 @@
+package cdc
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/bitmark-inc/autonomy-api/schema"
+	log "github.com/sirupsen/logrus"
+)
+
+type CovidSource string
+
+const (
+	CDSDaily                  CovidSource = "dailyFile"
+	CDSDailyHTTP              CovidSource = "dailyHttp"
+	CDSTimeseriesLocationFile CovidSource = "timeSeriesLocationFile"
+	CDSTimeseriesByDateFile   CovidSource = "timeSeriesByDateFile"
+)
+
+type CDS struct {
+	Country     string
+	Level       string
+	CDSDataType CovidSource
+	DataFile    *os.File
+	URL         string
+	Result      []schema.CDSData
+}
+
+func (c *CDS) Run() (int, error) {
+	data, err := dataFromURL(c.URL)
+	if nil != err {
+		return 0, err
+	}
+	count := 0
+	updateRecords := []schema.CDSData{}
+	sourceData := make([]interface{}, 0)
+
+	err = json.Unmarshal(data, &sourceData)
+
+	if err != nil {
+		fmt.Println("ParseDailyOnline error:", err)
+		return 0, err
+	}
+
+	for _, value := range sourceData {
+		record := schema.CDSData{}
+		object := value.(map[string]interface{})
+		name, ok := object["name"].(string)
+		if ok && len(name) > 0 && strings.Contains(name, c.Country) { // Country
+			record.Name = name
+		} else {
+			continue
+		}
+		record.City, _ = object["city"].(string)
+		record.County, _ = object["county"].(string)
+		record.State, _ = object["state"].(string)
+		record.CountryID, _ = object["countryId"].(string)
+		record.StateID, _ = object["stateId"].(string)
+		record.CountyID, _ = object["countyId"].(string)
+		record.Level, ok = object["level"].(string)
+
+		if ok && "" == record.Level {
```

Comment:
If `level` doesn't exist in the JSON object, also use the switch case to fill in the level value.